libc/src/__support/CPP/bit.h

```diff
@@ -269,14 +269,12 @@ first_trailing_one(T value) {
   return value == cpp::numeric_limits<T>::max() ? 0 : countr_zero(value) + 1;
 }
 
-/// Count number of 1's aka population count or hamming weight.
+/// Count number of 1's aka population count or Hamming weight.
 ///
 /// Only unsigned integral types are allowed.
-// TODO: rename as 'popcount' to follow the standard
-// https://en.cppreference.com/w/cpp/numeric/popcount
 template <typename T>
 [[nodiscard]] LIBC_INLINE constexpr cpp::enable_if_t<cpp::is_unsigned_v<T>, int>
-count_ones(T value) {
+popcount(T value) {
```

Comment:
Is there a reason we don't use `__builtin_popcount`? Pretty much every compiler supports it as far as I'm aware, or is there some benefit to relying on compiler optimizations for this.

Reply:
https://godbolt.org/z/M4sf9rr8Y seems the compiler can't completely figure it out. I'd recommend `libc_has_builitin` if we can.

Reply:
> Is there a reason we don't use __builtin_popcount?

128b, but we do use that builtin otherwise. See below (lines 284-295).

Reply:
Ah, didn't catch it in the context. Someone could probably implement it on `__128` in clang at least.

Reply:
I filed https://github.com/llvm/llvm-project/issues/82058 which resulted in https://github.com/llvm/llvm-project/commit/21d83324fbdbd91de0115d48f3b55979f57807b7.

We can use `__has_builtin` to use it for newer compilers.

Reply:
Neat, glad people are on top of things.